stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/viewmodel/MessageInputViewModel.kt

```diff
@@ -18,83 +20,121 @@ import io.getstream.chat.android.client.models.Message
 import io.getstream.chat.android.client.models.User
 import io.getstream.chat.android.client.utils.internal.toggle.ToggleService
 import io.getstream.chat.android.core.ExperimentalStreamChatApi
-import io.getstream.chat.android.livedata.ChatDomain
+import io.getstream.chat.android.offline.experimental.channel.state.ChannelState
+import io.getstream.chat.android.offline.experimental.extensions.asReferenced
+import io.getstream.chat.android.offline.experimental.extensions.globalState
 import io.getstream.chat.android.offline.extensions.setMessageForReply
+import kotlinx.coroutines.flow.combine
 import java.io.File
 
 /**
  * ViewModel class for MessageInputView. Responsible for sending and updating chat messages.
  * Can be bound to the view using the MessageInputViewModel.bindView function.
+ *
  * @param cid The full channel id, i.e. "messaging:123".
- * @param chatDomain Entry point for all livedata & offline operations.
  * @param chatClient Entry point for most of the chat SDK
  */
 public class MessageInputViewModel @JvmOverloads constructor(
     private val cid: String,
-    private val chatDomain: ChatDomain = ChatDomain.instance(),
     private val chatClient: ChatClient = ChatClient.instance(),
 ) : ViewModel() {
+
+    /**
+     * Holds information about the current channel and is actively updated.
+     */
+    public val channelState: ChannelState =
```

Comment:
Should we make it public? If yes, should we also expose other properties that can be accessed through `channelState`?